proposals/0000-initial-extension-categorization.md

```diff
@@ -0,0 +1,244 @@
+| DeriveFunctor              | Stable         | Stable                  |
+| DeriveGeneric              | Stable         | Stable                  |
+| DeriveLift                 | Stable         | Stable                  |
+| DeriveTraversable          | Stable         | Stable                  |
+| DerivingStrategies         | Stable         | Stable                  |
+| DerivingVia                | Stable         | Stable                  |
+| DisambiguateRecordFields   | Stable         | Stable                  |
+| EmptyCase                  | Stable         | Stable                  |
+| EmptyDataDecls             | Stable         | Stable                  |
+| EmptyDataDeriving          | Stable         | Stable                  |
+| ExistentialQuantification  | Stable         | Stable                  |
+| ExplicitForAll             | Stable         | Stable                  |
+| ExtendedDefaultRules       | Stable         | Stable                  |
+| FlexibleContexts           | Stable         | Stable                  |
+| FlexibleInstances          | Stable         | Stable                  |
+| ForeignFunctionInterface   | Stable         | Stable                  |
+| GADTs                      | Stable         | Stable                  |
+| GADTSyntax                 | Stable         | Stable                  |
+| GeneralizedNewtypeDeriving | Stable         | Stable                  |
+| HexFloatLiterals           | Stable         | Stable                  |
+| ImplicitParams             | Stable         | Stable                  |
+| ImplicitPrelude            | Stable         | Stable                  |
+| ImportQualifiedPost        | Stable         | Stable                  |
+| InstanceSigs               | Stable         | Stable                  |
+| InterruptibleFFI           | Stable         | Stable                  |
+| KindSignatures             | Stable         | Stable                  |
+| LambdaCase                 | Stable         | Stable                  |
+| LiberalTypeSynonyms        | Stable         | Stable                  |
+| MagicHash                  | Stable         | Stable                  |
+| MonadComprehensions        | Stable         | Stable                  |
+| MonomorphismRestriction    | Stable         | Stable                  |
+| MultiParamTypeClasses      | Stable         | Stable                  |
+| MultiWayIf                 | Stable         | Stable                  |
+| NamedFieldPuns             | Stable         | Stable                  |
+| NamedWildCards             | Stable         | Stable                  |
+| NumDecimals                | Stable         | Stable                  |
+| NumericUnderscores         | Stable         | Stable                  |
+| OverloadedLabels           | Stable         | Stable                  |
+| OverloadedLists            | Stable         | Stable                  |
+| OverloadedRecordDot        | Stable         | Stable                  |
```

Comment:
My impression is that `OverloadedRecordDot` (or perhaps just record extensions in general) have received quite a lot of changes and proposals recently. Perhaps @adamgundry knows more?

Reply:
I think `OverloadedRecordDot` has been fairly stable. Recently I generalised it (well, `HasField`) to support representation polymorphism, and there's a proposal to slightly widen the syntax at #668, but I don't expect any of that to break existing code.

The situation is slightly different with some of the other record extensions, e.g. `OverloadedRecordUpdate` is very unstable, while `DuplicateRecordFields` and `NoFieldSelectors` are somewhat in the middle, but that's reflected in the current categorisation.

Reply:
Either way, I've moved it to uncategorized for now.